### PR TITLE
[Core] Add tests to quaternions

### PR DESCRIPTION
**Pull request by joaquinirazabal**
Tests added to quaternion operations

### CODE REVIEW

File: kratos/tests/cpp_tests/utilities/test_quaternion.cpp [hunk 1]
```diff
@@ -0,0 +1,297 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
```
Comment (loumalouomega) [inline kratos/tests/cpp_tests/utilities/test_quaternion.cpp]: I have not done that :P

Comment (loumalouomega) [inline kratos/tests/cpp_tests/utilities/test_quaternion.cpp]:
```suggestion
//  Main authors:    Joaquín Irazabal Gonzalez
```

Comment (joaquinirazabal) [reply kratos/tests/cpp_tests/utilities/test_quaternion.cpp]: Thanks! Copy-paste issues :S

**Comment by maceligueta** [comment]
Many thanks @joaquinirazabal ! Great stuff. As this goes to the core, I will let the @KratosMultiphysics/technical-committee review it.

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**